controllers/users.js

```diff
@@ -0,0 +1,85 @@
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+
+const User = require('../models/users');
+const { JWT_SECRET_KEY } = require('../config');
+const NotFoundError = require('../errors/notFoundError');
+const ConflictError = require('../errors/conflictError');
+
+module.exports.getUsers = (req, res) => {
+  User.find({})
+    .then((users) => res.send({ data: users }))
+    .catch(() => res.status(500).send({ message: 'Произошла ошибка' }));
+};
+
+module.exports.createUser = (req, res) => {
+  const {
+    name, about, avatar, email, password,
+  } = req.body;
+
+  User.countDocuments({ email })
+    .then((count) => {
+      if (count) throw new ConflictError('Пользователь с таким email уже существует в базе');
+
+      bcrypt.hash(password, 10)
+        .then((hash) => User.create({
+          name,
+          about,
+          avatar,
+          email,
+          password: hash,
+        }))
+        .then((user) => res.send({ data: user.omitPrivate() }));
+    })
+    .catch((err) => {
+      const statusCode = err.statusCode || 500;
+      const message = statusCode === 500 ? 'Произошла ошибка' : err.message;
+      res.status(statusCode).send({ message });
+    });
```

Comment:
Нужно исправить: использовать единый `.catch()` блок для вложенных промисов.
Сейчас любой запрос с ошибкой в данных приведёт к тому, что исключение не будет обработано и клиент не получит ответ.
Обработка промисов работает только если дочерний промис возвращается:
```suggestion
      return bcrypt.hash(password, 10)
        .then((hash) => User.create({
          name,
          about,
          avatar,
          email,
          password: hash,
        }))
        .then((user) => res.send({ data: user.omitPrivate() }));
    })
    .catch((err) => {
      const statusCode = err.statusCode || 500;
      const message = statusCode === 500 ? 'Произошла ошибка' : err.message;
      res.status(statusCode).send({ message });
    });
```